### ISSUE TITLE
Attempt to get ES data using an invalid token in SiPixelPhase2DigiToClusterCUDA

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#36604]
Workflow 39434.502 step 3 is failing on slc7_ppc64le_gcc11 since CMSSW_12_3_X_2021-12-23-2300
```
----- Begin Fatal Exception 24-Dec-2021 07:02:22 CET-----------------------
An exception of category 'InvalidESGetToken' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 2 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Prefetching for module PrimaryVertexMonitor/'pixelPVMonitor'
   [3] Prefetching for module PixelVertexProducer/'pixelVertices'
   [4] Prefetching for module PixelTrackProducer/'pixelTracks'
   [5] Prefetching for module CAHitQuadrupletEDProducer/'pixelTracksHitQuadruplets'
   [6] Prefetching for module HitPairEDProducer/'pixelTracksHitDoublets'
   [7] Prefetching for module SeedingLayersEDProducer/'pixelTracksSeedLayers'
   [8] Prefetching for module SiPixelRecHitFromCUDA/'siPixelRecHitsPreSplitting@cuda'
   [9] Prefetching for module SiPixelRecHitCUDA/'siPixelRecHitsPreSplittingCUDA'
   [10] Calling method for module SiPixelPhase2DigiToClusterCUDA/'siPixelClustersPreSplittingCUDA'
Exception Message:
Attempted to get data using an invalid token of type ESGetToken<TrackerGeometry,TrackerDigiGeometryRecord>.
Please call consumes to properly initialize the token.
----- End Fatal Exception -------------------------------------------------
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_ppc64le_gcc11/CMSSW_12_3_X_2021-12-23-2300/pyRelValMatrixLogs/run/39434.502_TTbar_14TeV+2026D88_Patatrack_PixelOnlyGPU+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal/step3_TTbar_14TeV+2026D88_Patatrack_PixelOnlyGPU+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal.log#/

**Comment by makortel** [comment]
assign reconstruction, heterogeneous

**Comment by makortel** [comment]
I'm wondering why this is failing only in PowerPC IBs. Maybe the reason is that the PPC machine has GPUs
```
%MSG-i CUDAService:  (NoModuleName) 24-Dec-2021 07:00:45 CET pre-events
CUDA runtime version 11.4, driver version 11.4, NVIDIA driver version 440.33.01
CUDA device 0: Tesla P100-SXM2-16GB (sm_60)
CUDA device 1: Tesla P100-SXM2-16GB (sm_60)
CUDA device 2: Tesla P100-SXM2-16GB (sm_60)
CUDA device 3: Tesla P100-SXM2-16GB (sm_60)
%MSG
```
whereas for amd64 this workflow is not being tested in GPU_X IBs?

**Comment by makortel** [comment]
FYI @cms-sw/trk-dpg-l2 @AdrianoDee

**Comment by mmusich** [comment]
I think this can be closed, as it was solved by #36607, right?

**Comment by makortel** [comment]
+heterogeneous
